### ISSUE TITLE
unable to expand properties within a dict

### ISSUE THREAD

**Issue by r631269** [vim-vdebug/vdebug#312]
My issue's symptoms sound similar to #83 , but for python. I can inspect and expand variables, including lists and dicts, but if I try to expand anything within a dict, I get a `NameError`
For example, I can expand `user_pool_config.properties` (dict) and I see that `user_pool_config.properties['AutoVerifiedAttributes']` is a list with 1 element, but when I try to expand that, I get `- [ERROR] NameError: name 'AutoVerifiedAttributes' is not defined`. However, if I select `user_pool_config.properties['AutoVerifiedAttributes']` and run `<leader>e` it shows the value (which is, in turn, not expandable either)

I have patched my dbgp/client.py as suggested in :help. It looks like I'm at revision 4c6a7ca of vdebug

**Comment by BlackIkeEagle** [comment]
@r631269 Do you have a fix for this? could you be so kind to provide a PR? Could you verify if this is still present in v2?

**Comment by r631269** [comment]
I don't know how to fix it, so I don't have a PR
Where could I find instructions for getting v2 working?

**Comment by BlackIkeEagle** [comment]
@r631269 Ok I misinterpreted your line
> I have patched my dbgp/client.py.

and I've just merged v2 in master, it requires python 3 support in vim, for the rest its installable the same way you installed vdebug before

**Comment by BlackIkeEagle** [comment]
@r631269 could you provide a sample of what you are debugging and how you are debugging?

**Comment by r631269** [comment]
I finally got python3 support in macvim (I think), but I am unable to test, or debug anything at the moment. I'm seeing 

```
Vdebug will wait for a connection in the background
Found connection, starting debugger
Error starting Vdebug: <class 'UnicodeDecodeError'>
```

#330 looks relevant, but I'm already at revision `9c95ba44` and still seeing this problem

**Comment by BlackIkeEagle** [comment]
@r631269 could you share your vim configuration?

**Comment by r631269** [comment]
```  
  " vdebug
  if !exists('g:vdebug_options')
    let g:vdebug_options = {}
  endif
  let g:vdebug_options["break_on_open"]=0
  let g:vdebug_options["path_maps"] = {
  \  "/usr/src/app": "/Users/r631269/Documents/repos/stack-formation"
  \}
  let g:vdebug_options["port"] = 9001
```

**Comment by r631269** [comment]
https://gist.github.com/r631269/cae7892a1ef90871142401bf27e020ed

**Comment by BlackIkeEagle** [comment]
@r631269 I'm merging an update in master, could you re-test if the dict expanding works fine?

**Comment by r631269** [comment]
I did a `BundleInstall!`, which updated vdebug's revision to 5e1d2634, but I'm still getting the error listed above. It seems to happen as soon as vdebug connects to the debugger.

**Comment by BlackIkeEagle** [comment]
@r631269 are you on linux, mac, ... ? Since I'm having 0 issues with your vimrc now I'm kindof lost here how to solve your issue

**Comment by r631269** [comment]
I'm on a mac

**Comment by r631269** [comment]
Here's what I have in my ~/vdebug.log, in case that shows anything useful
```
- [Debug] {Wed 11 2018 09:30:46} Setting buffer for DebuggerWatch: ['']
- [Debug] {Wed 11 2018 09:30:46} Setting buffer for DebuggerStack: ['']
- [Debug] {Wed 11 2018 09:30:46} Setting buffer for DebuggerStatus: ['']
- [Info] {Wed 11 2018 09:30:46} Closing the connection
- [Debug] {Wed 11 2018 09:30:46} Command: stop -i 1
- [Debug] {Wed 11 2018 09:30:46} Response: <?xml version="1.0" encoding="utf-8"?>
<response xmlns="urn:debugger_protocol_v1" command="stop" status="stopped" reason="ok" transaction_id="1">application terminated</response>
- [Debug] {Wed 11 2018 09:30:46} Closing the socket
- [Debug] {Wed 11 2018 09:30:46} Closing the socket
- [Debug] {Wed 11 2018 09:30:46} Creating hidden buffer: ['']
- [Debug] {Wed 11 2018 09:30:46} Creating hidden buffer: ['']
- [Debug] {Wed 11 2018 09:30:46} Creating hidden buffer: ['']
- [Debug] {Wed 11 2018 09:30:46} Creating hidden buffer: [' 11004   | line        | /Users/r631269/Documents/repos/stack-formation/src/generate.py:160']
- [Debug] {Wed 11 2018 09:30:46} Creating hidden buffer: []
- [Debug] {Wed 11 2018 09:30:46} Creating hidden buffer: []
- [Debug] {Wed 11 2018 09:30:46} Closing the socket
```

**Comment by iamdonut** [comment]
@r631269 Perhaps try clearing your listchars setting (`set listchars=`) before starting a debugging session. I'm experiencing the same error and that's working for me as a quick fix.

**Comment by r631269** [comment]
thanks @iamdonut, made some progress. I do need listchars during development, but maybe I can do some vimscripting to disable them when the debugger starts or something. Should I file separate issue?

Now, back to the original issue, given a nested dict object 
```
obj = {
  "subobj": {
    "foo": 3
  }
}
``` 
expanding `subobj` in the debuggerwatch window results in the following
```
 Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/r631269/.vim/bundle/vdebug/pythonx/vdebug/debugger_interface.py", line 72, in handle_return_keypress
    return self.event_dispatcher.by_position(self.session_handler)
  File "/Users/r631269/.vim/bundle/vdebug/pythonx/vdebug/event.py", line 543, in by_position
    return event.run()
  File "/Users/r631269/.vim/bundle/vdebug/pythonx/vdebug/event.py", line 148, in run
    context_res = self.api.property_get(name)
  File "/Users/r631269/.vim/bundle/vdebug/pythonx/vdebug/dbgp.py", line 364, in property_get
    ContextGetResponse)
  File "/Users/r631269/.vim/bundle/vdebug/pythonx/vdebug/dbgp.py", line 263, in send_cmd
    return res_cls(msg, cmd, args, self)
  File "/Users/r631269/.vim/bundle/vdebug/pythonx/vdebug/dbgp.py", line 128, in __init__
    Response.__init__(self, response, cmd, cmd_args, api)
  File "/Users/r631269/.vim/bundle/vdebug/pythonx/vdebug/dbgp.py", line 30, in __init__
    self.__parse_error()
  File "/Users/r631269/.vim/bundle/vdebug/pythonx/vdebug/dbgp.py", line 50, in __parse_error
    raise DBGPError(msg_el.text, code)
vdebug.dbgp.DBGPError: ("NameError: name 'subobj' is not defined", '300')
```

**Comment by BlackIkeEagle** [comment]
@r631269 I have not found immediate issues running on macOS with your vimrc, using PHP (xdebug) in combination with vdebug:

https://herecura.eu/files/recording-2018-04-21_14.28.44.mp4

**Comment by BlackIkeEagle** [comment]
@r631269 could you provide me with a sample project where you encounter this issue? are you debugging python? or node?

**Comment by r631269** [comment]
Python. Sure, packaged it as a docker image so the debug client is more portable. Here's an example project:

[Dockerfile](https://github.com/r631269/python-debug-sample/blob/deleteme/Dockerfile) for reference. Image includes patched komodo, as instructed by vdebug :help

Place sample.py in a local directory
```
obj = {
  "subobj": {
    "foo": 3
  }
}
print(obj)
```

Run `docker run --rm -it -w /usr/src/app -v $(pwd):/usr/src/app r631269/python-debug-sample bash` from same directory as sample.py 

Run `python -S /opt/komodo/pydbgp -d docker.for.mac.localhost:9001 sample.py` from within the image, while vdebug is waiting for a connection

Expand `obj` in the watch window, then attempt to expand `subobj`

**Comment by BlackIkeEagle** [comment]
@r631269 thanks I can reproduce the issue now, the request sent from vdebug is incorrect, if its executed manually subobj is reachable. I'm going to search for a solution

**Comment by BlackIkeEagle** [comment]
@r631269 with the fix I just pushed I can expand all properties in your example

Thank you very much for providing this super helpfull minimal example.
